### PR TITLE
Tidy up the Pyccel homepage

### PR DESCRIPTION
**Pull request by EmilyBourne**
# Remove obsolete folders:
- `obsolete`
- `samples` : examples are all out of date
- `doc` : examples are all out of date
- `bugs` : code works but headers are out of date
- `inprogress` : contains OpenMP ideas from before implementation
- `benchmarks` : moved to [pyccel/pyccel-benchmarks](https://github.com/pyccel/pyccel-benchmarks) repository a year ago

# Remove obsolete files:
- `performance.md` : superseded by results from [pyccel/pyccel-benchmarks](https://github.com/pyccel/pyccel-benchmarks) repository
- `travis.yml` : unused and out of date

# Tidy up README:
- Avoid using relative links to ensure all links continue working on PyPi
- Use a raw link for the image so it displays on PyPi
- Move installation instructions to user documentation
- Add information about how to cite the JOSS paper
- Add CONTRIBUTING file to list of user documentation
- Update the image/link to the tests to point at `devel` branch tests not `master` tests

# Other
- Rename `tutorial` to `docs` so GitHub can find CONTRIBUTING.md file
- Fix deprecated warning check in tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,340 +14,57 @@ The aim of **Pyccel** is to provide a simple way to generate automatically, para
...
-brew install lapack
-brew install open-mpi
-brew install libomp
-```
-
-This requires that the Command Line Tools (CLT) for Xcode are installed.
+-   [Quick-start Guide](https://github.com/pyccel/pyccel/blob/devel/docs/quickstart.md)
 
-### Windows
+-   [Installation](https://github.com/pyccel/pyccel/blob/devel/docs/installation.md)
```
Comment (yguclu) [inline README.md:40]: I think that the installation is not just part of the user documentation, as it also describes editable installs. And in any case installing the library is the starting point for most people who might want to try Pyccel.

I suggest we write a very short "Installation" section before the "User Documentation" section, with just a few lines explaining that some requirements are needed and a link to `installation.md` for all the details.

File: README.md [hunk 2]
```diff
@@ -14,340 +14,57 @@ The aim of **Pyccel** is to provide a simple way to generate automatically, para
...
 
-Support for Windows is still experimental, and the installation of all requirements is more cumbersome.
-We recommend using [Chocolatey](https://chocolatey.org/) to speed up the process, and we provide commands that work in a git-bash sh.
-In an Administrator prompt install git-bash (if needed), a Python3 distribution, and a GCC compiler:
+-   [Contributing](https://github.com/pyccel/pyccel/blob/devel/docs/CONTRIBUTING.md)
```
Comment (yguclu) [inline README.md:42]: I think that "Contributing" also deserves a section, which can use the same approach as the "Installation" section.

**Review by yguclu — CHANGES_REQUESTED** [review]
Excellent job @EmilyBourne! I have just a few comments

File: README.md [hunk 3]
```diff
@@ -14,340 +14,57 @@ The aim of **Pyccel** is to provide a simple way to generate automatically, para
...
+If you are eager to try Pyccel out, we recommend reading our [quick-start guide](https://github.com/pyccel/pyccel/blob/devel/docs/quickstart.md).
 
-## Table of contents
+## Citing Pyccel
 
--   [User Documentation](#User-documentation)
+If Pyccel has been significant in your research, and you would like to acknowledge the project in your academic publication, we would ask that you cite the following paper:
 
--   [Developer Documentation](#Developer-documentation)
+Bourne et al., (2023). Pyccel: a Python-to-X transpiler for scientific high-performance computing. Journal of Open Source Software, 8(83), 4991, https://doi.org/10.21105/joss.04991
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply README.md]: Fixed 4bb1c908

File: README.md (hunk 2 above)
Comment (EmilyBourne) [reply README.md:64]: Fixed [4300001](https://github.com/pyccel/pyccel/pull/1438/commits/4300001043315b0f1954ae8327c8abfad239e3a5)  + [492d13f](https://github.com/pyccel/pyccel/pull/1438/commits/492d13f7f35ec90ff7fe6b41cbf20bc4bfd347a6)

File: README.md (hunk 1 above)
Comment (EmilyBourne) [reply README.md:62]: Fixed [4300001](https://github.com/pyccel/pyccel/pull/1438/commits/4300001043315b0f1954ae8327c8abfad239e3a5) + [afe96d6](https://github.com/pyccel/pyccel/pull/1438/commits/afe96d692efb18c31b85efb75d4c0b608d105bb6)

**Review by yguclu — APPROVED** [review]

**Pull request merged**